### PR TITLE
Joined Pitch Data to SQL Table

### PR DESCRIPTION
**Pull request by ewega**
- Pitch Dataframe is now joined to the one being sent to the SQL table

- Created the checkJoin function to check if a join worked

- Created the getEngine function to try for different engine connections depending on testing status
_Note: when testing, data is sent to localhost rather than our AWS database_

**HOW TO TEST**

- `cd audio-analysis`
- Make sure your localhost is setup (turn on Apache and SQL).
- Make sure _climactic_test_ in your localhost is empty
- Run `python main.py log` option 3
- Check database to ensure data is correctly input

### CODE REVIEW

File: audio-analysis/lib/etl/sqlConnection.py [hunk 1]
```diff
@@ -3,14 +3,34 @@
...
+        except:
+            print("\n ERROR: Issues connecting to your localhost \n")
+            
+    else:
+        try:
+            engine = create_engine('mysql+pymysql://' + db_username + ':' + db_password + '@climactic-test.cmikkru8vljn.us-east-1.rds.amazonaws.com:3306/climactic_test')
+            engine.connect()
+            print("\nYou are connected to ", engine, "\n")
+        except:
+            print("\n ERROR: Issues connecting to the AWS database \n")
```
Comment (mattlianje) [inline audio-analysis/lib/etl/sqlConnection.py:28]: Nice touch

File: audio-analysis/lib/etl/sqlConnection.py [hunk 2]
```diff
@@ -3,14 +3,34 @@
...
 # MAKE SURE YOU HAVE A DATABASE CALLED 'climactic_test'
 
-engine = create_engine('mysql+pymysql://' + db_username + ':' + db_password + '@climactic-test.cmikkru8vljn.us-east-1.rds.amazonaws.com:3306/climactic_test')
+
+def getEngine(isTest):
+    if isTest:
+        try:
+            engine = create_engine('mysql+pymysql://root:root@localhost/climactic_test')
+            engine.connect()
+            print("\nYou are connected to ", engine, "\n")
```
Comment (mattlianje) [inline audio-analysis/lib/etl/sqlConnection.py:14]: This idea is better than outputing to a csv. いいね<3

File: audio-analysis/lib/etl/sqlConnection.py [hunk 3]
```diff
@@ -3,14 +3,34 @@
 from secrets import db_username, db_password
 
 #### SQL CONNECTION INFO ####
-# Currently working on localhost
 # MAKE SURE YOU HAVE A DATABASE CALLED 'climactic_test'
 
-engine = create_engine('mysql+pymysql://' + db_username + ':' + db_password + '@climactic-test.cmikkru8vljn.us-east-1.rds.amazonaws.com:3306/climactic_test')
+
+def getEngine(isTest):
```
Comment (tjlam) [inline audio-analysis/lib/etl/sqlConnection.py:9]: Beautiful 🐐

File: audio-analysis/lib/etl/dfHelper.py [hunk 4]
```diff
@@ -24,3 +24,9 @@ def customLeftJoin(dfLeft, dfRight, joinCondition):
             joinedDf.set_value(index, 'amplitude_peak', True)
 
     return joinedDf
+
+def checkJoin(dfLeft, dfRight, joinDescription):
+    if (dfLeft.shape[0] == dfRight.shape[0]):
+        print('Join on "', joinDescription ,'" successful!')
+    else:
+        print('Join on "', joinDescription ,'" failed ...')
```
Comment (tjlam) [inline audio-analysis/lib/etl/dfHelper.py:32]: new line here

File: audio-analysis/lib/etl/dfHelper.py [hunk 5]
```diff
@@ -24,3 +24,9 @@ def customLeftJoin(dfLeft, dfRight, joinCondition):
             joinedDf.set_value(index, 'amplitude_peak', True)
 
     return joinedDf
+
+def checkJoin(dfLeft, dfRight, joinDescription):
```
Comment (tjlam) [inline audio-analysis/lib/etl/dfHelper.py:28]: Maybe we could make this return true or false so we can stop if the join didn't happen correctly?

File: audio-analysis/main.py [hunk 6]
```diff
@@ -29,23 +31,22 @@ def analyzeVideoSound(url, tag):
...
-        #Export video breakdown to SQL
-        df3.to_sql("test_table", con=engine, if_exists='append')
+    df1 = pd.DataFrame(video.word_list)
+    # We do not need two end_time_s_x and end_time_s_y columns :)
+    df2 = pd.DataFrame(video.amplitude_list).drop(columns=['end_time_s'])
+    join_condition = ['start_time_s', 'url']
+    print(df1)
+    print(df2)
+    df3 = dfHelper.customLeftJoin(df1, df2, join_condition)
+    dfHelper.checkJoin(df3, df1, "Words - Amplitude")
```
Comment (tjlam) [inline audio-analysis/main.py:41]: If checkJoin returns false do we want to stop? or keep joining?

File: audio-analysis/lib/etl/sqlConnection.py [hunk 7]
```diff
@@ -3,14 +3,34 @@
...
+    if isTest:
+        try:
+            engine = create_engine('mysql+pymysql://root:root@localhost/climactic_test')
+            engine.connect()
+            print("\nYou are connected to ", engine, "\n")
+        except Exception:
```
Comment (tjlam) [inline audio-analysis/lib/etl/sqlConnection.py:15]: Should we fail on Exception? whats the difference between `'mysql+pymysql://root@localhost/climactic_test'` and `'mysql+pymysql://root:root@localhost/climactic_test'`?

**Review by tjlam — CHANGES_REQUESTED** [review]
Nice clean up work, a few questions but thats all

File: audio-analysis/main.py [hunk 8]
```diff
@@ -12,6 +12,8 @@
 # Using this Trump Video for testing
 trump_video = "https://www.youtube.com/watch?v=JZRXESV3R74"
 
+engine = getEngine(TESTING)
```
Comment (mattlianje) [inline audio-analysis/main.py:15]: good idea))

**Review by mattlianje — APPROVED** [review]
Very nice)

File: audio-analysis/lib/etl/sqlConnection.py (hunk 7 above)
Comment (mattlianje) [reply audio-analysis/lib/etl/sqlConnection.py:15]: we need to get to the bottom of default user for MySQL db on local. Somehow for Wega it is password "NO" and for me password "YES"

Comment (tjlam) [reply audio-analysis/lib/etl/sqlConnection.py:15]: Ahh I see. I feel kinda behind in all the sql db setup. Maybe next meeting we can go over it?

Comment (mattlianje) [reply audio-analysis/lib/etl/sqlConnection.py:15]: yes. It really is a two click thing))

File: audio-analysis/lib/etl/dfHelper.py (hunk 5 above)
Comment (mattlianje) [reply audio-analysis/lib/etl/dfHelper.py:28]: agreed.

File: audio-analysis/main.py (hunk 6 above)
Comment (mattlianje) [reply audio-analysis/main.py:41]: I also ~agreed~ agree with stopping ... the real fix will be another PR though. I like checkJoin a lot ... right now it is in the wrong place ... but lets leave it be. In the next 2-3 days will:
- make a test.py for all our unit tests using ye olde `unittest`
- use asserts the proper way in this test.py file to throw errors nicely.

**Review by tjlam — APPROVED** [review]

**Pull request merged**